core/description/resource.go

```diff
@@ -203,6 +203,70 @@ func (r *resourceRevision) Username() string {
 	return r.Username_
 }
 
+func (r *resourceRevision) merge(args ResourceRevisionArgs) (*resourceRevision, error) {
+	// Check fields match where set.
+	mkErr := func(field string) error {
+		return errors.Errorf("%s mismatch for revision %d", field, args.Revision)
+	}
+	checkStr := func(field, a, b string) error {
+		if a != "" && b != "" && a != b {
```

Comment:
If a is not empty and b is empty, doesn't that count as a mismatch? As in we are trying to overwrite a previously set value with empty?